astroquery/svo_fps/core.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import requests
+import io
+
+from astropy import units as u
+from astropy.io.votable import parse_single_table
+# The VOTables fetched from SVO contain only single table element, thus parse_single_table
+
+from ..query import BaseQuery
+
+__all__ = ['SvoFpsClass', 'SvoFps']
+
+FLOAT_MAX = np.finfo(np.float64).max
+
+
+class SvoFpsClass(BaseQuery):
+    SVO_MAIN_URL = 'http://svo2.cab.inta-csic.es/theory/fps/fps.php'
+
+    def data_from_svo(self, query, cache=True,
+                      error_msg='No data found for requested query'):
+        """Get data in response to the query send to SVO FPS
+
+        Parameters
+        ----------
+        query : dict
+            Used to create a HTTP query string i.e. send to SVO FPS to get data.
+            In dictionary, specify keys as search parameters (str) and
+            values as required. List of search parameters can be found at
+            http://svo2.cab.inta-csic.es/theory/fps/fps.php?FORMAT=metadata
+        error_msg : str, optional
+            Error message to be shown in case no table element found in the
+            responded VOTable. Use this to make error message verbose in context
+            of the query made (default is 'No data found for requested query')
+        cache : bool
+            Cache the results?  Defaults to True
+
+        Returns
+        -------
+        astropy.table.table.Table object
+            Table containing data fetched from SVO (in response to query)
+        """
+        response = self._request("GET", self.SVO_MAIN_URL, params=query,
+                                 cache=cache)
+        response.raise_for_status()
+        votable = io.BytesIO(response.content)
+        try:
+            return parse_single_table(votable).to_table()
```

Comment:
This function seems to return a table with byte strings instead of regular strings. Ideally you would convert any strings that can be ascii strings into ascii strings, but if not this needs to be clearly documented, because for example the following does not work, although it feels like it should:
```
index = SvoFps.get_filter_index()
galex = index[index["Facility"] == "GALEX"]
```
and instead you have to do:
```
galex = index[index["Facility"] == b"GALEX"]
```

Reply:
This can't be done, apparently.  the votable reader won't accept string
```
  File "/home/adam/anaconda3/lib/python3.7/site-packages/astropy/io/votable/table.py", line 185, in parse_single_table
    votable = parse(source, **kwargs)
  File "/home/adam/anaconda3/lib/python3.7/site-packages/astropy/utils/decorators.py", line 521, in wrapper
    return function(*args, **kwargs)
  File "/home/adam/anaconda3/lib/python3.7/site-packages/astropy/io/votable/table.py", line 167, in parse
    config=config, pos=(1, 1)).parse(iterator, config)
  File "/home/adam/anaconda3/lib/python3.7/site-packages/astropy/io/votable/tree.py", line 3505, in parse
    for start, tag, data, pos in iterator:
TypeError: expected bytes, str found
```

Reply:
Boooooooo! OK, not something we can fix.